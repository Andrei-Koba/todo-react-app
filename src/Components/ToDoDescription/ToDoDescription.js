import React from 'react';
import { PropTypes } from 'prop-types';
import styles from './ToDoDescription.module.scss';

export default function ToDoDescription({ isCompleted, text }) {
  return (
    <span className={`${styles.itemText} ${isCompleted ? styles.crossed : undefined}`}>{text}</span>
  );
}

ToDoDescription.propTypes = {
  text: PropTypes.string.isRequired,
  isCompleted: PropTypes.bool.isRequired,
};
