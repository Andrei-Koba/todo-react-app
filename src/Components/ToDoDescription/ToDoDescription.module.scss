.itemText {
    margin-left: 15px;
    line-height: 25px;
     
    &.crossed {
        color: lightgray;
        text-decoration: line-through;
    }
}