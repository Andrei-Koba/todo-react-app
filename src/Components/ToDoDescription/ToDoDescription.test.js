import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ToDoDescription from './ToDoDescription';

describe('ToDoDescription', () => {
  it.each([true, false])('renders correctly, isCompleted="%p"', async (isCompleted) => {
    const actual = render(<ToDoDescription isCompleted={isCompleted} text="testText" />);
    expect(actual).toMatchSnapshot();
  });
});
