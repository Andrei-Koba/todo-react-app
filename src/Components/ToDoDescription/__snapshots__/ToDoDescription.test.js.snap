// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoDescription renders correctly, isCompleted="false" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="itemText undefined"
      >
        testText
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="itemText undefined"
    >
      testText
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ToDoDescription renders correctly, isCompleted="true" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="itemText crossed"
      >
        testText
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="itemText crossed"
    >
      testText
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
